File: 7. Reverse Integer/ReverseInteger.c
int reverse(int x){
    // Limit variable.
    int negLimit = -2147483647;
    int posLimit = 2147483647;
    
    // overflow integer.
    if (x < negLimit || x > posLimit) return 0;

    // result variable.
    long res = 0;

    // reverse loop.
    while (x != 0) {
        int rem = x % 10;
        res = (res * 10) + rem;
        x /= 10;
    }

    // overflow integer.
    if (res < negLimit || res > posLimit) return 0;
    return res;
}

File: 7. Reverse Integer/ReverseInteger.java
class Solution {
    public int reverse(int x) {
        int r = 0; 
        long result = 0;
        boolean flag = false;
        int len = 0;
        // -2147483648 to 2147483647
            if(x < 0) {
                flag = true;
                x = -x;
            }

            while(x > 0) {
                r = x % 10;
                result = result * 10 + r;
                x = x / 10;
                len++;
            }

        if (-2147483648 > result || 2147483647 < result)
            return 0;
        if (flag)
            return (int)-result;
        return (int)result;
    }
}
